File: flutter_app/lib/screens/success_screen.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import '../theme/app_colors.dart';

class SuccessScreen extends StatelessWidget {
  final String type;
  final String title;

  const SuccessScreen({
    super.key,
    this.type = 'complete',
    this.title = 'Task',
  });

  @override
  Widget build(BuildContext context) {
    final config = _typeConfig();

    return Scaffold(
      backgroundColor: config.bg,
      body: SafeArea(
        child: Column(
          children: [
            Expanded(
              child: Center(
                child: SingleChildScrollView(
                  padding: const EdgeInsets.all(32),
                  child: Column(
                    mainAxisSize: MainAxisSize.min,
                    children: [
                      Container(
                        width: 120,
                        height: 120,
                        decoration: BoxDecoration(
                          color: config.iconBg,
                          shape: BoxShape.circle,
                          border: Border.all(color: config.iconBorder, width: 6),
                        ),
                        child: Icon(config.icon, size: 64, color: config.iconColor),
                      ),
                      const SizedBox(height: 32),
                      Text(
                        config.status,
                        style: TextStyle(
                          fontSize: 16,
                          fontWeight: FontWeight.w700,
                          letterSpacing: 1.5,
                          color: config.iconBg,
                        ),
                      ),
                      const SizedBox(height: 8),
                      Text(
                        config.heading,
                        style: const TextStyle(
                          fontSize: 36,
                          fontWeight: FontWeight.w900,
                          color: AppColors.heading,
                        ),
                        textAlign: TextAlign.center,
                      ),
                      const SizedBox(height: 12),
                      Text(
                        config.message(title),
                        style: const TextStyle(
                          fontSize: 18,
                          fontWeight: FontWeight.w500,
                          color: AppColors.mutedText,
                        ),
                        textAlign: TextAlign.center,
                      ),
                      const SizedBox(height: 40),
                      LayoutBuilder(
                        builder: (context, constraints) {
                          final buttons = [
                            _navButton(
                              context, 'Return to Home', '/home',
                              primary: true,
                              color: config.iconBg,
                            ),
                            const SizedBox(width: 16, height: 16),
                            _navButton(
                              context, "Return to Today's Plan", '/todays-plan',
                              primary: false,
                              color: config.iconColor,
                            ),
                          ];

                          if (constraints.maxWidth > 500) {
                            return Row(children: [
                              Expanded(child: buttons[0]),
                              buttons[1],
                              Expanded(child: buttons[2]),
                            ]);
                          }
                          return Column(
                            crossAxisAlignment: CrossAxisAlignment.stretch,
                            children: buttons,
                          );
                        },
                      ),
                    ],
                  ),
                ),
              ),
            ),
            Container(
              padding: const EdgeInsets.all(16),
              child: Text(
                'Action Recorded Securely',
                style: TextStyle(
                  fontSize: 16,
                  fontWeight: FontWeight.w700,
                  color: config.iconColor.withValues(alpha: 0.7),
                ),
                textAlign: TextAlign.center,
              ),
            ),
          ],
        ),
      ),
    );
  }

  Widget _navButton(
    BuildContext context,
    String label,
    String path, {
    bool primary = false,
    Color? color,
  }) {
    if (primary) {
      return SizedBox(
        width: double.infinity,
        height: 60,
        child: ElevatedButton(
          onPressed: () => context.go(path),
          style: ElevatedButton.styleFrom(
            backgroundColor: color ?? AppColors.primaryAction,
            foregroundColor: AppColors.white,
            shape: RoundedRectangleBorder(
              borderRadius: BorderRadius.circular(20),
            ),
          ),
          child: Text(
            label,
            style: const TextStyle(fontSize: 18, fontWeight: FontWeight.w700),
          ),
        ),
      );
    }
    return SizedBox(
      width: double.infinity,
      height: 60,
      child: OutlinedButton(
        onPressed: () => context.go(path),
        style: OutlinedButton.styleFrom(
          foregroundColor: AppColors.heading,
          side: const BorderSide(color: AppColors.primary, width: 3),
          shape: RoundedRectangleBorder(
            borderRadius: BorderRadius.circular(20),
          ),
        ),
        child: Text(
          label,
          style: const TextStyle(fontSize: 18, fontWeight: FontWeight.w700),
        ),
      ),
    );
  }

  _SuccessConfig _typeConfig() {
    switch (type) {
      case 'snooze':
        return _SuccessConfig(
          bg: AppColors.warningBg,
          iconBg: AppColors.warningDark,
          iconBorder: AppColors.warningDark,
          iconColor: AppColors.warningLight,
          icon: Icons.snooze,
          status: 'SNOOZED',
          heading: 'Reminder Snoozed',
          message: (t) => '$t reminder snoozed for 1 hour.',
        );
      case 'postpone':
        return _SuccessConfig(
          bg: AppColors.purpleBg,
          iconBg: AppColors.purpleLight,
          iconBorder: AppColors.purpleLight,
          iconColor: AppColors.caregiver,
          icon: Icons.schedule,
          status: 'POSTPONED',
          heading: 'Task Postponed',
          message: (t) => '$t has been moved to a later time.',
        );
      default:
        return _SuccessConfig(
          bg: AppColors.successBg,
          iconBg: AppColors.successLight,
          iconBorder: AppColors.successLight,
          iconColor: AppColors.success,
          icon: Icons.check_circle,
          status: 'COMPLETE',
          heading: 'Task Complete',
          message: (t) => '$t marked as complete.',
        );
    }
  }
}

class _SuccessConfig {
  final Color bg;
  final Color iconBg;
  final Color iconBorder;
  final Color iconColor;
  final IconData icon;
  final String status;
  final String heading;
  final String Function(String) message;

  const _SuccessConfig({
    required this.bg,
    required this.iconBg,
    required this.iconBorder,
    required this.iconColor,
    required this.icon,
    required this.status,
    required this.heading,
    required this.message,
  });
}

File: react-native-app/src/screens/SuccessScreen.js
import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { Ionicons } from '@expo/vector-icons';
import Colors from '../theme/colors';

const TYPE_CONFIG = {
  complete: { icon: 'checkmark-circle', color: Colors.success, bg: Colors.successBg, heading: 'Task Complete!', message: 'Great job! Your task has been marked as done.' },
  snooze: { icon: 'time', color: Colors.warning, bg: Colors.warningBg, heading: 'Snoozed', message: 'We\'ll remind you again later.' },
  postpone: { icon: 'calendar', color: Colors.primaryAction, bg: Colors.blueBg, heading: 'Postponed', message: 'This task has been rescheduled.' },
};

export default function SuccessScreen({ navigation, route }) {
  const type = route.params?.type || 'complete';
  const title = route.params?.title || '';
  const config = TYPE_CONFIG[type] || TYPE_CONFIG.complete;

  return (
    <SafeAreaView style={styles.safe}>
      <View style={styles.container}>
        <View style={[styles.iconCircle, { backgroundColor: config.bg }]}>  
          <Ionicons name={config.icon} size={64} color={config.color} />
        </View>
        <Text style={styles.heading}>{config.heading}</Text>
        {title ? <Text style={styles.title}>{title}</Text> : null}
        <Text style={styles.message}>{config.message}</Text>

        <View style={styles.actions}>
          <TouchableOpacity
            style={styles.primaryButton}
            onPress={() => navigation.reset({ index: 0, routes: [{ name: 'Home' }] })}
          >
            <Text style={styles.primaryText}>Return to Home</Text>
          </TouchableOpacity>
          <TouchableOpacity
            style={styles.secondaryButton}
            onPress={() => navigation.reset({ index: 0, routes: [{ name: 'TodaysPlan' }] })}
          >
            <Text style={styles.secondaryText}>Return to Today's Plan</Text>
          </TouchableOpacity>
        </View>

        <View style={styles.securityFooter}>
          <Ionicons name="shield-checkmark" size={16} color={Colors.mutedText} />
          <Text style={styles.securityText}>Your data is stored securely on this device</Text>
        </View>
      </View>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  safe: { flex: 1, backgroundColor: Colors.pageBg },
  container: { flex: 1, justifyContent: 'center', alignItems: 'center', padding: 24 },
  iconCircle: {
    width: 120, height: 120, borderRadius: 60,
    justifyContent: 'center', alignItems: 'center', marginBottom: 24,
  },
  heading: { fontSize: 30, fontWeight: '900', color: Colors.heading, textAlign: 'center' },
  title: { fontSize: 20, fontWeight: '700', color: Colors.mutedText, marginTop: 8, textAlign: 'center' },
  message: { fontSize: 18, color: Colors.mutedText, textAlign: 'center', marginTop: 8, lineHeight: 28 },
  actions: { width: '100%', marginTop: 32, gap: 12 },
  primaryButton: {
    backgroundColor: Colors.primaryAction, borderRadius: 20,
    paddingVertical: 18, alignItems: 'center',
  },
  primaryText: { fontSize: 20, fontWeight: '700', color: Colors.white },
  secondaryButton: {
    backgroundColor: Colors.white, borderRadius: 20,
    paddingVertical: 18, alignItems: 'center',
    borderWidth: 3, borderColor: Colors.primaryAction,
  },
  secondaryText: { fontSize: 20, fontWeight: '700', color: Colors.primaryAction },
  securityFooter: { flexDirection: 'row', alignItems: 'center', gap: 8, marginTop: 32 },
  securityText: { fontSize: 14, color: Colors.mutedText },
});
